controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -2565,13 +2610,43 @@ int bml_internal::get_restricted_channels(uint8_t *restricted_channels, const st
     return (iRet);
 }
 
-int bml_internal::wfca_controller(const char *cmd, char *ret_buf, int ret_buf_size)
+int bml_internal::wfa_ca_controller(BML_CTX ctx, const char *command, int command_len,
+                                    const char **first_reply, BML_WFA_CA_CB second_reply_cb)
 {
-    // TODO: implement function
-    return BML_RET_OK;
+    //CMDU message
+    auto request =
+        message_com::create_vs_message<beerocks_message::cACTION_BML_WFA_CA_CONTROLLER_REQUEST>(
+            cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR) << "Failed building cACTION_BML_WFA_CA_CONTROLLER_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->alloc_command(command_len);
+    auto command_tuple = request->command(0);
+    if (!std::get<0>(command_tuple)) {
+        LOG(ERROR) << "command buffer has failed!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    char *command_buff = (char *)(&(std::get<1>(command_tuple)));
+    string_utils::copy_string(command_buff, command, command_len);
+
+    m_cbWfaCaSecondReply = second_reply_cb;
+
+    int result;
+    int iRet = send_bml_cmdu(result, request->get_action_op());
+    if (!result) {
+        return (-BML_RET_OP_FAILED);
+    }
+
+    *first_reply = m_wfa_ca_controller_first_reply.c_str();
+
+    return iRet;
 }
 
-int bml_internal::wfca_agent(const char *cmd, char *ret_buf, int ret_buf_size)
+int bml_internal::wfa_ca_agent(const char *cmd, char *ret_buf, int ret_buf_size)
```

Comment:
Why isn't there cmd_len argument here? It is only OK not to include length of a const char *buffer if it is hard coded in the code, for example `const char *message ="send1905_cmdu,1234"` - AFAIK this is not the case, the string is received from the testbed through a TCP socket.

Reply:
I only changed this function name. Currently, it is not used.
On Agent certification tasks we will probably change/remove it...